frontend/src/component/HomePage.tsx

```diff
@@ -53,8 +54,13 @@ export default function Home() {
         }
     }, [topic, setEdges, setNodes])
 
+    const handleSubmitAdd: MouseEventHandler<HTMLButtonElement> = (event) => {
+        window.location.assign("/topic/add");
```

Comment:
This reloads the whole app. It is better to use client side routing: Here is some issues brought up by BingChat: 
Using window.location.assign() may lead to problems when you use ReactRouter in your app, depending on your use case. Some possible problems are:

   Using window.location.assign() will cause a full page reload, which may affect the performance and user experience of your app
    Using window.location.assign() will not update the history object that ReactRouter uses to manage navigation. [This may cause unexpected behavior when using methods like history.push() or history.goBack()]
    Using window.location.assign() may not work well with nested routes or dynamic routes that ReactRouter supports.

A better way to navigate to a different URL within your app is to use the Link component or the useHistory hook from ReactRouter. [These methods will update the history object and avoid reloading the page]. 
If you want to navigate to an external URL, you can use a regular anchor tag (<a>) instead of window.location.assign().